staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -68,6 +68,10 @@ func (k *notRegisteredErr) Error() string {
 	return fmt.Sprintf("no kind %q is registered for version %q in scheme %q", k.gvk.Kind, k.gvk.GroupVersion(), k.schemeName)
 }
 
+func (k *notRegisteredErr) Is(err error) bool {
+	return IsNotRegisteredError(err)
```

Comment:
this is only checking the error type, it should be checking gvk / target / t match as well